src/test/java/org/example/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,138 @@
+package org.example.bookstore.controller;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import org.example.bookstore.dto.book.BookDto;
+import org.example.bookstore.exception.EntityNotFoundException;
+import org.example.bookstore.security.JwtUtil;
+import org.example.bookstore.service.BookService;
+import org.example.bookstore.util.TestUtil;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@Sql(scripts = "classpath:database/remove-all-books-from-table.sql",
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Mock
+    private BookService bookService;
+    @MockBean
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @Transactional
```

Comment:
Why do you need this annotation?

Reply:
When i added sql scripts on class layer some tests don't work. How i understand java try to compile a several tests simultaneously and a few tests dont work so a i added Transational so that they are executed one by one

On screenshot i demonstrate how test work if i remove Transational
![image](https://github.com/user-attachments/assets/5d8dc7fb-d467-432c-9b5d-58d5db894cf6)

Reply:
Try to use the annotation above your class. Let’s discover how it works. And remove Transactional  ;) 
```
@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)
```